.github/workflows/full_stack_tests.yml

```diff
@@ -189,105 +189,136 @@ jobs:
+
         # TODO(#22667): We are running this step twice, as snapshot gets deleted if 2nd attempt passes.
-      - name: Uploading diff screenshots as an artifact
-        if: ${{ failure() }}
+      - name: Upload diff screenshots as an artifact
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
-          name: diff-snapshots-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
+          name: diff-snapshots-desktop-attempt-1-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
           # Note: The path must be kept in sync with the one specified in
           # core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts
           path: /home/runner/work/oppia/oppia/core/tests/puppeteer-acceptance-tests/diff-snapshots
+
+      # Desktop Test (Attempt 2)
       - name: Run Desktop Acceptance Test ${{ matrix.suite.name }} (Attempt 2 with screen recording)
-        if: ${{ failure() }}
+        id: desktop-test-attempt-2
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         run: >
           set -o pipefail;
           VIDEO_RECORDING_IS_ENABLED=1
           xvfb-run -a --server-args="-screen 0, 1920x1080x24"
           python -m scripts.run_acceptance_tests
           --skip-build --suite=${{ matrix.suite.name }} --prod_env --server_log_level=info
           | tee desktop_acceptance_test_retry_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
+
       - name: Upload desktop acceptance retry test logs
-        if: ${{ failure() }}
+        if: ${{ steps.desktop-test-attempt-2.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
           name: desktop-acceptance-test-retry-logs-${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}
           path: desktop_acceptance_test_retry_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
-      - name: Uploading Desktop test recordings as Artifacts
-        if: ${{ failure() }}
+      
+      - name: Upload desktop test recordings as an artifact
+        # We are checking for failure of desktop-test-attempt-1 and not desktop-test-attempt-2 because
+        # because we want to upload video recording regardless of whether desktop-test-attempt-2 passes or fails.
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
           name: desktop-acceptance-test-video-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
           path: /home/runner/work/oppia/oppia_full_stack_test_video_recordings/acceptance
-      - name: Uploading generated test to angular module mapping as an artifact
-        if: ${{ failure() }}
+
+        # TODO(#22667): We are running this step twice, as snapshot gets deleted if 2nd attempt passes.
```

Comment:
Now that we have different names (desktop-attempt-1 and desktop-attempt-2), is this still a concern?

Reply:
It don't think it's a concern. I also think we can close the issue #22667. As, deleting diff-screenshot when second run starts is an expected behaviour. If we don't it will create confusion between diff-screenshot of attempt 1 and attempt 2.

Reply:
Done.

Reply:
I have added video recording for attempt 1. Should I keep video recording in attempt 2 or should I delete it?

Reply:
You don't need to do a video recording in attempt 2. (Don't record in the first place, don't save images -- attempt 2 is just for testing whether the error is permanent.)